usr/lib/api/socket_client.c

```diff
@@ -303,13 +305,13 @@ static void *event_thread(void *arg)
 #endif
 
     /* Enable cancellation */
-    pthread_setcancelstate(PTHREAD_CANCEL_ENABLE, &oldstate);
-    pthread_setcanceltype(PTHREAD_CANCEL_DEFERRED, &oldtype);
     cleanup.anchor = anchor;
 #if OPENSSL_VERSION_PREREQ(3, 0)
     cleanup.prev_libctx = prev_libctx;
 #endif
     pthread_cleanup_push(event_thread_cleanup, &cleanup);
+    pthread_setcanceltype(PTHREAD_CANCEL_DEFERRED, &oldtype);
+    pthread_setcancelstate(PTHREAD_CANCEL_ENABLE, &oldstate);
```

Comment:
Don't know if this makes a difference: when looking for pthread examples on the Internet, they always first set the cancelstate and then the canceltype. Like it was before in removed lines 306+307. Did you change the order intentionally?

Reply:
Right, I changed the order intentionally because it also sounds more logical to me to first tell how to cancel it and second allowing to cancel it. I originally had it the other way round because I probably simply copied it from the examples.... If it does not really make a difference, then thats even better.